File: backend/src/main/java/com/petconect/backend/model/VetService.java
package com.petconect.backend.model;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class VetService {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private String description;
    private double price;
    private Long ownerId;
    private String ownerName;
    private String ownerLocation;
    private String ownerPhone;
    private String ownerCrmv;
    private String operatingHours;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
} 

File: lib/models/vet_service.dart
import 'package:json_annotation/json_annotation.dart';

part 'vet_service.g.dart';

@JsonSerializable()
class VetService {
  final int? id;
  final String name;
  final String? description;
  final double price;
  final int? ownerId;
  final String? ownerName;
  final String? ownerLocation;
  final String? ownerPhone;
  final String? ownerCrmv;
  final String? operatingHours;
  final String? createdAt;
  final String? updatedAt;

  VetService({
    this.id,
    required this.name,
    this.description,
    required this.price,
    this.ownerId,
    this.ownerName,
    this.ownerLocation,
    this.ownerPhone,
    this.ownerCrmv,
    this.operatingHours,
    this.createdAt,
    this.updatedAt,
  });

  factory VetService.fromJson(Map<String, dynamic> json) => _$VetServiceFromJson(json);
  Map<String, dynamic> toJson() => _$VetServiceToJson(this);
} 